### ISSUE TITLE
Impact of Primitive Cell Choice on Calculated Material Properties with easyunfold.

### ISSUE THREAD

**Issue by hongyi-zhao** [SMTG-Bham/easyunfold#48]
Hello easyunfold developer,

I've been noticed that various material properties, including elastic, dielectric and piezoelectric properties, the forms of the results are affected by the choice of the primitive cell, as discussed [here](https://mp.weixin.qq.com/s?__biz=MzI2OTQ4OTExOA==&mid=2247485044&idx=2&sn=80178d0bda6de43e3a5e4a47a72aa158&chksm=eadec0c5dda949d31853dafa37ae593b96f872990bf7b7ada0cbb89bf5e64bbc66903d8794e4&scene=21#wechat_redirect).

Given that different tools can generate varying forms of a material's primitive cell, I'm curious about the potential implications when using easyunfold for subsequent calculations and processing. Specifically, I'm interested in understanding: How does choosing the primitive cell form (as obtained from different computational tools) affect the results obtained with easyunfold for unfolding band structures or calculating material properties?

As an example, we can take the POSCARs shown on the website mentioned above to conduct our discussion: [POSCAR_.zip](https://github.com/SMTG-Bham/easyunfold/files/14798679/POSCAR_.zip)

Thank you for your time and support.

Best regards,
Zhao

**Comment by zhubonan** [comment]
> I've been noticed that various material properties, including elastic, dielectric and piezoelectric properties, the forms of the results are affected by the choice of the primitive cell, as discussed [here](https://mp.weixin.qq.com/s?__biz=MzI2OTQ4OTExOA==&mid=2247485044&idx=2&sn=80178d0bda6de43e3a5e4a47a72aa158&chksm=eadec0c5dda949d31853dafa37ae593b96f872990bf7b7ada0cbb89bf5e64bbc66903d8794e4&scene=21#wechat_redirect).

They would be dependent on the choice of the primitive cell, in particular, the *orientation* of the primitive cell (in factor, the *orientation* of the underlying crystal) as these properties are often computed for the Cartesian coordinates (e.g. the lab frame). If you rotate the crystal, these properties would rotate accordantly. 

I think it is important to separate *differences* in the primitive cells in to two cases: 1. The difference in crystal orientation. 2. the difference in the choice of lattice vectors with same underlying crystal orientation. For things like elastic, dielectric and piezoelectric properties (which are tensors), it is the first one that makes the difference. Although the second one can also affect the calculation as it changes the k-point sampling, the FFT grid etc, but they should not make much effect if the calculation is properly converged. 

> How does choosing the primitive cell form (as obtained from different computational tools) affect the results obtained with easyunfold for unfolding band structures or calculating material properties?

For things like band structure, the paths are generated using the reciprocal lattice vectors as the basis, so the orientation of the crystal in the cartesian frame does not matter, but these paths are generated for *specific* standardised primitive cells. Behind the scene, one often uses `spglib`'s routine to find such standardised primitive cell and use that it to define the paths (in cases of sumo/seekpath/pymatgen, I think). This standardised primitive cell may not be the same as your input "primitive cell". Now, choices of lattice vectors matter, but the orientation of the crystal does not. If this standardised primitive cell is simply a rotated version of your input "primitive" cell (e.g. mismatch only in crystal orientation) then it is fine to use your input primitive cell, otherwise you *must*  use the standardised primitive cell for which the path is generated for. 

`easyunfold` does not deal with the generation of paths - it simply reads in those generated by packages (e.g. sumo). However, it is nevertheless extremely important to ensure that the primitive cell passed is indeed the same standardised primitive cell for which the path is generated. Otherwise, one risk ending up with incorrect labels for the band structure (which, unfortunately, are not uncommon in the literature).

To summarise, physical properties should not depend on which cell you choose, but their *presentations* may. A band structure plot is just a slice through the reciprocal space along specific paths - it changes when you choose a different cell/path, but the underlying physics remain the same.
